src/main/scala/viper/silver/cfg/silver/SilverCfg.scala

```diff
@@ -82,6 +82,65 @@ class SilverCfg(val blocks: Seq[SilverBlock], val edges: Seq[SilverEdge], val en
                     entry: SilverBlock = entry,
                     exit: Option[SilverBlock] = exit): SilverCfg =
     SilverCfg(blocks, edges, entry, exit)
+
+  private def findJoinPoint(queueInit: Iterable[SilverBlock],
+                            visitedInit: Iterable[SilverBlock],
+                            // We never enqueue loopHeads which we already have seen.
+                            // This would lead to non-termination.
+                            loopHeadsSeen: Iterable[SilverBlock],
+                            getNext: SilverBlock => Iterable[SilverBlock])
+  : (Option[SilverBlock], mutable.Map[SilverBlock, SilverBlock]) = {
+
+    var queue = mutable.Queue.from(queueInit)
+    var visited: Vector[SilverBlock] = Vector.from(visitedInit)
+    val map = mutable.Map[SilverBlock, SilverBlock]()
+    var loopHeads: Vector[SilverBlock] = Vector.from(loopHeadsSeen)
```

Comment:
Type `Vector[SilverLoopHeadBlock]` seems more appropriate

Reply:
Yes, however this would require additional casting in later when `loopHeads` is used.

Reply:
Maybe pattern matching instead of isInstanceOf would yield a nice solution? But do as you deem fit, I don't have a strong opinion here.